File: src/dotnet/Ansight.Core/Pairing/Models/PairingChallenge.cs
namespace Ansight.Pairing.Models;

/// <summary>
/// Describes the host challenge material required to authenticate the pairing attempt.
/// </summary>
public sealed class PairingChallenge
{
    /// <summary>
    /// Algorithm identifier used for the challenge/proof flow.
    /// </summary>
    public required string Alg { get; set; }

    /// <summary>
    /// Challenge public key or equivalent challenge material exposed by the host.
    /// </summary>
    public required string ChallengePubKey { get; set; }

    /// <summary>
    /// Indicates whether proof is required on the first successful pair.
    /// </summary>
    public required bool RequireProofOnFirstPair { get; set; }
}

File: src/ios/Sources/AnsightKit/PairingChallenge.swift
import CryptoKit
import Foundation

public struct PairingChallenge: Sendable, Codable, Equatable {
    public var alg: String
    public var challengePubKey: String
    public var requireProofOnFirstPair: Bool

    public init(alg: String, challengePubKey: String, requireProofOnFirstPair: Bool) {
        self.alg = alg
        self.challengePubKey = challengePubKey
        self.requireProofOnFirstPair = requireProofOnFirstPair
    }
}
